SystemInformer/syssccpu.c

```diff
@@ -1903,3 +1925,87 @@ PPH_STRINGREF PhGetHybridProcessorType(
     return NULL;
 }
 #endif
+
+/**
+ * \brief Checks if a specific logical processor (core) in the system is parked.
+ *
+ * This function determines whether a specific logical processor in the system
+ * is parked, meaning it is temporarily unavailable for the thread scheduler.
+ * It utilizes CPU sets functionality available starting with Windows 10.
+ *
+ * \param[in] ProcessorIndex The index of the logical processor to check.
+ *
+ * \return TRUE if the specified logical processor is parked, FALSE otherwise or
+ * in case of failure.
+ */
+BOOLEAN PhIsCoreParked(
+    _In_ ULONG ProcessorIndex
+    )
+{
+    PSYSTEM_CPU_SET_INFORMATION cpuSets = NULL;
+    ULONG returnedLength;
+    HANDLE currentProcess = GetCurrentProcess();
```

Comment:
Current process is optional if we aren't interested in `AllocatedToTargetProcess`. Since we aren't interested in that field we should probably pass `NULL` instead?

Reply:
That's right. As of https://github.com/winsiderss/systeminformer/pull/2040/commits/4458096161c84278c458cbac55f57d1f73b636c1 we use `NtQuerySystemInformationEx` in which we don't pass a handle to the actual process (but we do pass a valid indirected handle as requested by the routine).